lib/fizzy/leb128.hpp

```diff
@@ -19,11 +22,14 @@ std::pair<T, const uint8_t*> leb128u_decode(const uint8_t* input, const uint8_t*
         if (input == end)
             throw parser_error("Unexpected EOF");
 
-        // TODO this ignores the bits in the last byte other than the least significant one
-        // So would not reject some invalid encoding with those bits set.
         result |= static_cast<T>((static_cast<T>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
+        {
+            if (*input != (result >> result_shift))
```

Comment:
In the signed case the expected side is masked with `0x7f`. Is that needed here too?

Reply:
No because right-shifing unsigned value produces 0 in the most significant bit anyway.